src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareInProcLanguageClient.cs

```diff
@@ -42,6 +43,17 @@ internal class LiveShareInProcLanguageClient : AbstractInProcLanguageClient
         public override string Name => ServicesVSResources.Live_Share_CSharp_Visual_Basic_Language_Server_Client;
 
         protected internal override VSServerCapabilities GetCapabilities()
-            => _defaultCapabilitiesProvider.GetCapabilities();
+        {
+            var experimentationService = Workspace.Services.GetRequiredService<IExperimentationService>();
+            var isLspEditorEnabled = experimentationService.IsExperimentEnabled(VisualStudioWorkspaceContextService.LspEditorFeatureFlagName);
+            // If the preview feature flag to turn on the LSP editor in local scenarios is on, advertise no capabilities for this Live Share
+            // LSP server as LSP requests will be serviced by the AlwaysActiveInProcLanguageClient in both local and remote scenarios.
+            if (isLspEditorEnabled)
+            {
+                return new VSServerCapabilities();
```

Comment:
This might not be a problem, but this isn't "no capabilities", it will still get didOpen notifications: https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FRemoteLanguage.Protocol%2FServerCapabilities.cs&version=GBdevelop&line=30&lineEnd=31&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents